### ISSUE TITLE
[Spike] Allow community admins to set or update chain nodes

### ISSUE THREAD

**Issue by dillchen** [hicommonwealth/commonwealth#5377]
## Description

This has been a blocker in the past when something breaks, we cut down on P0 (for specific communities) in case of chain node failures. There may be security implications for this so unclear how we want to handle this. From the product and growth side this is a pure desired feature.

## Acceptance Criteria
1. In Community admin panel, admin is able to update the chain node

For the spike
1. Security implications if any? (any mitigations that engineering is comfortable with -- possible solutions 1) allow users to specifically set it 2) allow user opt in to allow for custom chain node set by admin 3) just allow admins)

## Additional context
Other products allow users to set custom chain nodes, however that's done on a user by user basis, see image from screenshot it's polkadot apps

<img width="331" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/5334557/d8aba648-6860-462e-8d80-2015d219ad20">

**Comment by Rotorsoft** [comment]
After an initial discussion of requirements and primary concerns with @ianrowan, we reached an agreement aimed at achieving a balance between granting administrators more flexibility to configure their own nodes and addressing the security and reliability issues we've raised. Here's our proposed plan:

1. Administrators will have the option to add new nodes. However, we will implement an automated protocol verification step to ensure that the node aligns with the chain versions we support. Updates will only be accepted if all checks pass successfully.

2. Once accepted, any actions performed within that community will be accompanied by a warning sign. This warning will clearly indicate that the community is utilizing an unverified node and explicitly state our security concerns.

3. Our growth team will subsequently establish a manual verification process in collaboration with the community administrator. This process will be prioritized based on the community's importance, and the community will be labeled as a "verified community" once the verification is complete.

**Comment by jnaviask** [comment]
cc @dillchen @zakhap to review, then we can mark complete.

**Comment by Rotorsoft** [comment]
@jnaviask I think there were some discussion in Istanbul about simplifying the way we manage nodes, tagging @ianrowan and @timolegros

**Comment by dillchen** [comment]
I think the comment above that @Rotorsoft made makes full sense to me. What are the updates from Istanbul that need to be reviewed?

**Comment by zakhap** [comment]
This all makes sense to me, in regards to "verification". Is there a way we can distinguish between a "Verified Chain" and a "Verified Community"? Can we be more specific to the users _what exactly about their community is **unverified?**_

**Comment by Rotorsoft** [comment]
> This all makes sense to me, in regards to "verification". Is there a way we can distinguish between a "Verified Chain" and a "Verified Community"? Can we be more specific to the users _what exactly about their community is **unverified?**_

That should be "verified chain"

**Comment by zakhap** [comment]
Makes sense to me! What are the product/UX improvement conversations that were had in Istanbul?

**Comment by Rotorsoft** [comment]
tagging @timolegros  and @ianrowan - not sure if specifically related to this, but they were mentioning something about simplifying the way we deal with chains/nodes after they attended one of the events
